sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -3258,7 +3258,9 @@ class SQLConf extends Serializable with Logging {
   }
 
   def clear(): Unit = {
+    val reserved = settings.asScala.filterKeys(staticConfKeys.contains).toMap
     settings.clear()
+    settings.putAll(reserved.asJava)
```

Comment:
is this change necessary? static SQL configs can only be set with `SparkConf`. We can clear everything here, and the `RESET` command will add back the configs in `SparkConf`

Reply:
yes. it is not necessary.